plugins/hls-notes-plugin/src/Ide/Plugin/Notes.hs

```diff
@@ -69,20 +81,56 @@ findNotesRules recorder = do
 
     defineNoDiagnostics (cmapWithPrio LogShake recorder) $ \MkGetNotes _ -> do
         targets <- toKnownFiles <$> useNoFile_ GetKnownTargets
-        definedNotes <- catMaybes <$> mapM (\nfp -> fmap (HM.map (nfp,)) <$> use MkGetNotesInFile nfp) (HS.toList targets)
+        definedNotes <- catMaybes <$> mapM (\nfp -> fmap (HM.map (nfp,) . fst) <$> use MkGetNotesInFile nfp) (HS.toList targets)
         pure $ Just $ HM.unions definedNotes
 
+    defineNoDiagnostics (cmapWithPrio LogShake recorder) $ \MkGetNoteReferences _ -> do
+        targets <- toKnownFiles <$> useNoFile_ GetKnownTargets
+        definedReferences <- catMaybes <$> mapM (\nfp -> fmap (HM.map (fmap (nfp,)) . snd) <$> use MkGetNotesInFile nfp) (HS.toList targets)
```

Comment:
```suggestion
        definedReferences <- catMaybes <$> for (HS.toList targets) $ \nfp -> do
          (_, references) <- use MkGetNotesInFilie nfp
          pure $ HM.map (fmap (nfp,)) references
```
up to you, but I think making it less one-liner-y is easier to read. Even better to use `forMaybe` but we probably don't want to depend on `witherable`...